platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/SwirldState.java

```diff
@@ -102,6 +103,19 @@ default void preHandle(final Event event) {}
      */
     void handleConsensusRound(final Round round, final SwirldDualState swirldDualState);
 
+    /**
+     * Update the stake of the nodes in the address book
+     *
+     * @param configAddressBook the address book as loaded from config.txt. This address book may contain new nodes not
+     *                          present in the stateAddressBook. Must not be null.
+     * @param stateAddressBook  the address book loaded from the latest state on disk, or null if starting from
+     *                          genesis.  May be null.
+     * @return a copy of the configuration address book with updated stake.
+     */
+    default AddressBook updateStake(final AddressBook configAddressBook, final AddressBook stateAddressBook) {
```

Comment:
Is there a reason this method has to be part of `SwirldState`, as opposed to maybe in `AddressBook` directly?

Reply:
The application implements both SwirldState and SwirldMain.   I also found it confusing why there would be program logic in a State class, but there is other application logic bearing methods in the SwirldState.  SwirldState is where I was told to put it as part of the original design (see our notion doc.)